### PR TITLE
Avx debug

### PR DESCRIPTION
**Pull request by sarahsummerfield**
No longer segfaulting and test files have been standardized. All that's left is tweaking the actual calculation stuff to get the right numbers. I think this no longer constitutes an actual "debug", so it would be a good time to merge this back into the AVX feature branch.

### CODE REVIEW

File: src/DistanceCalculatorAVX.cpp [hunk 1]
```diff
@@ -364,11 +365,16 @@ double DistanceCalculator::newCalcDNA(int a, int b) {
         for (int j = 0; i < numOfInts && j < 31;
              i += 8) { // a maximum of 32 vectors allowed not to overflow the
                        // int the numbers are stored in
-            seq1 = *(__m256i *)&Achar[i];
-            seq2 = *(__m256i *)&Bchar[i];
 
-            gap1 = *(__m256i *)&Agap[i];
-            gap2 = *(__m256i *)&Bgap[i];
+            seq1 = _mm256_load_si256((__m256i*)&Achar[i]);
+            //seq1 = *(__m256i *)&Achar[i];
```
Comment (glesica) [inline src/DistanceCalculatorAVX.cpp]: Why'd you leave these commented? Can we just delete them?

File: src/DistanceCalculatorAVX.cpp [hunk 2]
```diff
@@ -377,45 +383,57 @@ double DistanceCalculator::newCalcDNA(int a, int b) {
...
         }
 
-        /*gather transversion counts*/
+        /*gather transversion counts:
+        Compute the absolute differences of packed unsigned 8-bit
+        integers in a and b, then horizontally sum each
+        consecutive 8 differences to produce four unsigned 16-bit
+        integers, and pack these unsigned 16-bit integers in the
+        low 16 bits of 64-bit elements in dst.
+        **in avx this gives 4 16-bit integers, not 2*/
```
Comment (glesica) [inline src/DistanceCalculatorAVX.cpp:392]: This comment is a little confusing. We're packing four 16-bit integers into a single 64-bit element? Or one per 64-bit element?

File: src/TreeBuilderManager.cpp [hunk 3]
```diff
@@ -99,10 +99,12 @@ std::string TreeBuilderManager::doJob() {
...
-            DistanceCalculator *distCalc = new DistanceCalculator(
-                seqs, (DistanceCalculator::AlphabetType)alphType,
-                (DistanceCalculator::CorrectionType)corrType,
-                seqReader->numSeqs, this->newDistanceMethod);
+            void *distCalcMem = _mm_malloc(sizeof(DistanceCalculator), 32);
+            DistanceCalculator *distCalc = static_cast<DistanceCalculator*>(distCalcMem);
+            new(distCalc) DistanceCalculator(
+                    seqs, (DistanceCalculator::AlphabetType)alphType,
+                    (DistanceCalculator::CorrectionType)corrType,
+                    seqReader->numSeqs, this->newDistanceMethod);
```
Comment (glesica) [inline src/TreeBuilderManager.cpp:107]: Do we still need to do this?

File: src/DistanceCalculatorAVX.cpp (hunk 1 above)
Comment (sarahsummerfield) [reply src/DistanceCalculatorAVX.cpp]: I didn't have a firm reason, other than I suspect we're going to have to make a lot of similar changes through the codebase and I find it helpful having the before and after when comparing to other areas... we don't need to keep them in, though!

File: src/DistanceCalculatorAVX.cpp (hunk 2 above)
Comment (sarahsummerfield) [reply src/DistanceCalculatorAVX.cpp:392]: It's packing 1 integer into the low 16 bits of each of the 64 bit blocks... Slightly confusing phrasing courtesy of the Intel Intrinsics guide.

File: src/TreeBuilderManager.cpp (hunk 3 above)
Comment (sarahsummerfield) [reply src/TreeBuilderManager.cpp:107]: I think we do, until/if we change all the field allocations to use aligned_alloc.

**Comment by sarahsummerfield** [comment]
@glesica I've pushed the changes that we discussed on Friday!

**Pull request merged**